File: Projeto_2/src/main.py
"""Script principal para demonstração de Polimorfismo no cálculo de prêmios."""

from models.cliente import Cliente
from models.seguros import SeguroAuto, SeguroVida, SeguroResidencial

def main():
    # 1. Criação do cliente e proteção de dados sensíveis
    cliente = Cliente(nome="Alan Turing", cpf="123.456.789-00")
    
    # 2. Instanciando a lista de seguros (diferentes tipos misturados)
    apolices = [
        SeguroAuto(titular=cliente, valor_base=1000.0, placa="ABC-1234", ano=2008),
        SeguroAuto(titular=cliente, valor_base=1000.0, placa="XYZ-9876", ano=2021),
        SeguroVida(titular=cliente, valor_base=2500.0, idade=65),
        SeguroVida(titular=cliente, valor_base=2500.0, idade=35),
        SeguroResidencial(titular=cliente, valor_base=1200.0, tipo_residencia="CASA"),
        SeguroResidencial(titular=cliente, valor_base=1200.0, tipo_residencia="APARTAMENTO")
    ]
    
    print(f"--- Relatório de Prêmios - Cliente: {cliente.nome} ---")
    
    # 3. POLIMORFISMO EM AÇÃO
    # O laço interage com a abstração `Seguro`. O sistema não sabe se é Vida,
    # Auto ou Residencial; ele apenas invoca `calcular_premio()` e cada objeto
    # responde com sua própria regra de negócio.
    for apolice in apolices:
        tipo_seguro = apolice.__class__.__name__
        premio_final = apolice.calcular_premio()
        print(f"Apolice: {tipo_seguro.ljust(18)} | Valor do Prêmio: R$ {premio_final:,.2f}")

if __name__ == "__main__":
    main()

File: Projeto_2/main.js
const Cliente = require('./src/models/Cliente');
const SeguroAuto = require('./src/models/SeguroAuto');
const SeguroVida = require('./src/models/SeguroVida');
const SeguroResidencial = require('./src/models/SeguroResidencial');

const cliente = new Cliente("Williams Torres", "123.456.789-00", "williams@email.com");

const auto = new SeguroAuto("Williams Torres", 1000.0, "ABC-1234", 2008);
const vida = new SeguroVida("Williams Torres", 500.0, 65);
const residencial = new SeguroResidencial("Williams Torres", 800.0, "CASA", 120.0);

cliente.adicionarSeguro(auto);
cliente.adicionarSeguro(vida);
cliente.adicionarSeguro(residencial);

console.log(`Cliente: ${cliente.nome}\n`);
for (const seguro of cliente.listarSeguros()) {
  console.log(seguro.toString());
}
